File: php/src/BL/BLEmpresa.php
<?php
/** Lógica de negocio de Empresas. Equivalente a BL/BLEmpresa.vb. */

require_once __DIR__ . '/../DA/DAEmpresa.php';
require_once __DIR__ . '/../DA/DAUsuario.php';

class BLEmpresa
{
    private DAEmpresa $da;

    public function __construct()
    {
        $this->da = new DAEmpresa();
    }

    public function CargarDepartamento(?string $ccod_cia = null): array
    {
        return $this->da->CargarDepartamento($ccod_cia);
    }

    public function CargarProvincia(string $id_departamento): array
    {
        return $this->da->CargarProvincia($id_departamento);
    }

    public function CargarDistrito(string $id_provincia): array
    {
        return $this->da->CargarDistrito($id_provincia);
    }

    public function CargarCompania(string $cod): array
    {
        return $this->da->CargarCompania($cod);
    }

    public function CargarCompanias(): array
    {
        return $this->da->CargarCompanias();
    }

    public function InsertarCompania(BEEmpresa $obj): bool
    {
        return $this->da->InsertarCompania($obj);
    }

    public function EditarCompania(BEEmpresa $obj): bool
    {
        return $this->da->EditarCompania($obj);
    }

    public function EliminarEmpresa(string $cod): bool
    {
        return $this->da->EliminarEmpresa($cod);
    }

    public function CargarCompaniasConBDValida(): array
    {
        return (new DAUsuario())->ConsultarEmpresasConBDValida();
    }
}

File: BL/BLEmpresa.vb
Public Class BLEmpresa

    Dim objDA As New DA.DAEmpresa()


    Public Function CargarDepartamento(ccod_cia As String)
        Return objDA.CargarDepartamento(ccod_cia)
    End Function

    Public Function CargarProvincia(id_departamento As String)
        Return objDA.CargarProvincia(id_departamento)
    End Function

    Public Function CargarDistrito(id_provincia As String)
        Return objDA.CargarDistrito(id_provincia)
    End Function
     
    'Public Function CargarTarifa()
    '    Return objDA.CargarTarifa()
    'End Function
     
    Public Function CargarCompania(cod As String)
        Return objDA.CargarCompania(cod)
    End Function


    Public Function CargarCompanias()
        Return objDA.CargarCompanias()
    End Function

    Public Function InsertarCompania(objBE As BE.BEEmpresa)
        Return objDA.InsertarCompania(objBE)
    End Function

    Public Function EditarCompania(objBE As BE.BEEmpresa)
        Return objDA.EditarCompania(objBE)
    End Function


    Public Function EliminarEmpresa(cod As String)
        Return objDA.EliminarEmpresa(cod)
    End Function

    Public Function CargarCompaniasConBDValida()
        Return New DA.DAUsuario().ConsultarEmpresasConBDValida()
    End Function

End Class
